lib/uknetdev/netdev.c

```diff
@@ -274,43 +301,71 @@ void uk_netdev_info_get(struct uk_netdev *dev,
 				      dev_info->max_tx_queues);
 }
 
-static const void *_netdev_einfo_get(struct uk_netdev *dev,
-				enum uk_netdev_einfo_type einfo)
-{
-	switch (einfo) {
-	case UK_NETDEV_IPV4_ADDR_STR:
-		if (dev->_einfo->ipv4_addr)
-			uk_pr_debug("ip: %s\n", dev->_einfo->ipv4_addr);
-		return dev->_einfo->ipv4_addr;
-	case UK_NETDEV_IPV4_MASK_STR:
-		if (dev->_einfo->ipv4_net_mask)
-			uk_pr_debug("netmask: %s\n", dev->_einfo->ipv4_net_mask);
-		return dev->_einfo->ipv4_net_mask;
-	case UK_NETDEV_IPV4_GW_STR:
-		if (dev->_einfo->ipv4_gw_addr)
-			uk_pr_debug("gateway: %s\n", dev->_einfo->ipv4_gw_addr);
-		return dev->_einfo->ipv4_gw_addr;
-	case UK_NETDEV_IPV4_DNS0_STR:
-		if (dev->_einfo->ipv4_dns0_addr)
-			uk_pr_debug("dns: %s\n", dev->_einfo->ipv4_dns0_addr);
-		return dev->_einfo->ipv4_dns0_addr;
-	default:
-		uk_pr_warn("Option %d not yet supported\n", einfo);
-	}
-	return NULL;
-}
-
-const void *uk_netdev_einfo_get(struct uk_netdev *dev,
+const char *uk_netdev_einfo_get(struct uk_netdev *dev,
 				enum uk_netdev_einfo_type einfo)
 {
 	UK_ASSERT(dev);
 	UK_ASSERT(dev->ops);
 	UK_ASSERT(dev->_data->state >= UK_NETDEV_UNCONFIGURED);
 
-	if (dev->_einfo)
-		return _netdev_einfo_get(dev, einfo);
-	else if (dev->ops->einfo_get)
+#if CONFIG_LIBUKNETDEV_EINFO_LIBPARAM
+	if (dev->_einfo) {
+		switch (einfo) {
+		case UK_NETDEV_IPV4_ADDR:
+			if (dev->_einfo->ip4.cidr ||
+			    (dev->ops->einfo_get &&
+			     dev->ops->einfo_get(dev, UK_NETDEV_IPV4_CIDR)))
+				return NULL; /* CIDR (overwrite) exists */
+			break;
+		case UK_NETDEV_IPV4_MASK:
+			if (dev->_einfo->ip4.cidr ||
+			    (dev->ops->einfo_get &&
+			     dev->ops->einfo_get(dev, UK_NETDEV_IPV4_CIDR)))
+				return NULL; /* CIDR (overwrite) exists */
+			break;
+		case UK_NETDEV_IPV4_CIDR:
+			if (dev->_einfo->ip4.cidr)
+				return dev->_einfo->ip4.cidr;
+			break;
+		case UK_NETDEV_IPV4_GW:
+			if (dev->_einfo->ip4.gw)
+				return dev->_einfo->ip4.gw;
+			break;
+		case UK_NETDEV_IPV4_DNS0:
+			if (dev->_einfo->ip4.dns0)
+				return dev->_einfo->ip4.dns0;
+			break;
+		case UK_NETDEV_IPV4_DNS1:
+			if (dev->_einfo->ip4.dns1)
+				return dev->_einfo->ip4.dns1;
+			break;
+		case UK_NETDEV_IPV4_HOSTNAME:
+			if (dev->_einfo->ip4.hostname)
+				return dev->_einfo->ip4.hostname;
+			break;
+		case UK_NETDEV_IPV4_DOMAIN:
+			if (dev->_einfo->ip4.domain)
+				return dev->_einfo->ip4.domain;
+			break;
+		default:
```

Comment:
```suggestion
		default:
		    break;
```

This will error on compile time.

Reply:
What is the error that you are receiving? I am unable to reproduce an error. It compiles fine for me with GCC 13.2.0.

Reply:
As we just discussed and confirmed, in fact, older GCC versions handle this differently and do not accept an empty `default` case. I'll push an update. Thanks for reporting this. 👍🏻